utils/etherspot.js

```diff
@@ -62,21 +74,22 @@ export async function investByAAWallet(investmentAmount, chosenToken) {
   // console.log("\x1b[33m%s\x1b[0m", `EtherspotWallet balances:`, balances);
 }
 
-async function getPortfolioHelper(portfolioName) {
+async function getPortfolioHelper(portfolioName, mappedProvider) {
   let portfolioHelper;
   if (portfolioName === "AllWeatherPortfolio") {
-    portfolioHelper = new AllWeatherPortfolio();
+    portfolioHelper = new AllWeatherPortfolio(mappedProvider);
   }
   await portfolioHelper.initialize();
   return portfolioHelper;
 }
 
 export class AllWeatherPortfolio {
-  constructor() {
+  constructor(mappedProvider) {
     // initializating sdk...
     const customBundlerUrl = "";
     this.primeSdk = new PrimeSdk(
-      { privateKey: process.env.NEXT_PUBLIC_WALLET_PRIVATE_KEY },
+      mappedProvider,
+      // { privateKey: process.env.NEXT_PUBLIC_WALLET_PRIVATE_KEY },
```

Comment:
error: `Error: invalid address or ENS name (argument="name", value=null, code=INVALID_ARGUMENT, version=contracts/5.7.0)`

Here's the `mappedProvider` that I got : `mappedProvider > Web3WalletProvider {type: 'Web3', address$: UniqueSubject, networkName$: UniqueSubject, web3: JsonRpcProvider}`